simple_graph.py

```diff
@@ -16,17 +16,25 @@ def __init__(self):
 
     def add_node(self, value):
         """Add node to Graph."""
-        self.container.setdefault(value, [])
+        if value in self.container:
```

Comment:
This conditional is unnecessary. dict.setdefault(val, {}) would leave "val" alone if it's already set to something